src/main/java/io/oneko/docker/DockerRegistryPolling.java

```diff
@@ -67,23 +72,41 @@
 	@Scheduled(fixedDelay = 20000, initialDelay = 20000)
 	protected void checkDockerForNewImages() {
 		if (!this.pollingInProgress.getAndSet(true)) {
+			log.trace("Starting polling job");
+			lastPollingJobStartDate = Instant.now();
 			final Mono<List<ProjectMesh>> meshesMono = projectMeshRepository.getAll().
 					collectList();
 			final Mono<List<Project>> projectsMono = projectRepository.getAll()
 					.collectList();
-			meshesMono.zipWith(projectsMono)
-					.subscribe(pair -> this.checkDockerForNewImages(pair.getT1(), pair.getT2()));
+			this.currentPollingJob = meshesMono.zipWith(projectsMono)
+					.flatMap(pair -> this.checkDockerForNewImages(pair.getT1(), pair.getT2()))
+					.subscribe();
+		} else if (shouldPollingJobBeTimedOut() || this.currentPollingJob.isDisposed()) {
+			log.info("Timing out polling job.");
+			this.currentPollingJob.dispose();
+			this.currentPollingJob = null;
+			setPollingJobToNotRunning();
 		} else {
 			log.debug("Skipping job because another polling job is still running.");
 		}
 	}
 
+	private boolean shouldPollingJobBeTimedOut() {
+		final Duration pollingTimeout = Duration.ofMinutes(5);
```

Comment:
5 Minutes seems quite arbitrary.

Reply:
Because it is. I can make it less static though.